routing/missioncontrol_store.go

```diff
@@ -222,38 +266,118 @@ func deserializeResult(k, v []byte) (*paymentResult, error) {
 
 // AddResult adds a new result to the db.
 func (b *missionControlStore) AddResult(rp *paymentResult) error {
-	return kvdb.Update(b.db, func(tx kvdb.RwTx) error {
-		bucket := tx.ReadWriteBucket(resultsKey)
+	b.queueMx.Lock()
```

Comment:
Don't need to return an error here anymore? Surprised the linter isn't angry.

Reply:
indeed! done